sites/all/modules/custom/jdt_user_notes/jdt_user_notes.module

```diff
@@ -39,81 +37,244 @@ function jdt_user_notes_permission() {
+      $form = drupal_get_form('jdt_user_notes_form', $article_nid, $note_id);
     }
+
+    return $form;
+  }
+}
+
+/**
+ * Callback function for displaying article notes.
+ */
+function jdt_user_notes_display($article_nid) {
+  global $user;
+
+  ctools_include('modal');
+  ctools_modal_add_js();
+
+  $notes = jdt_user_notes_read_article($article_nid->nid);
+
+  $output = array(
+    '#type' => 'container',
+    '#attributes' => array(
+      'class' => array('notes-container'),
+    ),
+  );
+
+  foreach ($notes as $note) {
+    $note_container = 'note-container-' . $note['note_id'];
+    $output[$note_container] = array(
+      '#type' => 'container',
+      '#attributes' => array(
+        'class' => array('note-container'),
+      ),
+    );
+
+    $output[$note_container]['note-text'] = array(
+      '#type' => 'container',
+      '#attributes' => array(
+        'class' => array('note-text'),
+      ),
+    );
+
+    $output[$note_container]['note-text']['text'] = array(
+      '#type' => 'markup',
+      '#markup' => $note['note_text'],
+    );
+
+    $output[$note_container]['node-title'] = array(
+      '#type' => 'container',
+      '#attributes' => array(
+        'class' => array('node-title'),
+      ),
+    );
+
+    $output[$note_container]['node-title']['title'] = array(
+      '#type' => 'markup',
+      '#markup' => $article_nid->title,
+    );
+
+    $output[$note_container]['node-author'] = array(
+      '#type' => 'container',
+      '#attributes' => array(
+        'class' => array('node-author'),
+      ),
+    );
+
+    $output[$note_container]['node-author']['author'] = array(
+      '#type' => 'markup',
+      '#markup' => $user->name,
+    );
+
+    $output[$note_container]['edit-link'] = array(
+      '#type' => 'container',
+      '#attributes' => array(
+        'class' => 'edit-link',
+      ),
+    );
+
+    $link = 'node/' . $note['article_nid'] . '/edit/' . $note['note_id'] . '/nojs';
+    $output[$note_container]['edit-link']['link'] = array(
+      '#type' => 'markup',
+      '#markup' => l(t('Edit Note'), $link, array(
+        'attributes' => array(
+          'class' => 'ctools-use-modal',
+        ),
+      )),
+    );
+
+  }
+
+  return $output;
+}
+
+/**
+ * Validates both the add and edit forms.
+ *
+ * Ensures note_title and note_body do not exceed 255 characters.
+ */
+function jdt_user_notes_validate($form, &$form_state) {
+  if (strlen($form_state['input']['note_title']) > 254) {
+    form_set_error('note_title', t('Your note title must be less than 255 
+      characters.'));
+  }
+  if (strlen($form_state['input']['note_body']) > 254) {
+    form_set_error('note_body', t('Your note body must be less than 255 
+      characters.'));
   }
 }
 
 /**
- * Custom handler for jdt_user_notes_edit().
+ * Access callback function for notes page.
  */
-function jdt_user_notes_edit($note_id, $article_nid) {
-  $check = jdt_user_notes_read($note_id);
+function jdt_user_notes_access($article_nid) {
```

Comment:
This also needs check for the permission that you set up in people -> permissions. Make sure you export these permissions.